src/client/pages/ChangePassword.tsx

```diff
@@ -17,203 +17,251 @@ import { Main } from '@/client/layouts/Main';
 import { Header } from '@/client/components/Header';
 import { Footer } from '@/client/components/Footer';
 import { CsrfFormField } from '@/client/components/CsrfFormField';
-import { PasswordWeakMessage } from '@/client/components/PasswordWeakMessage';
-import { PasswordLengthMessage } from '@/client/components/PasswordLengthMessage';
 import { space } from '@guardian/src-foundations';
 import { css } from '@emotion/react';
-import {
-  ErrorValidationResult,
-  LengthValidationResult,
-  PasswordValidationLongMessage,
-  PasswordValidationResult,
-  validatePasswordLength,
-} from '@/shared/lib/PasswordValidation';
-import { ValidationStyling } from '@/client/styles/PasswordValidationStyles';
+
 import sha1 from 'js-sha1';
 import { PasswordInput } from '@/client/components/PasswordInput';
 import { FieldError } from '@/shared/model/ClientState';
+import { ChangePasswordErrors } from '@/shared/model/Errors';
 import AwesomeDebouncePromise from 'awesome-debounce-promise';
 
 type ChangePasswordProps = {
   submitUrl: string;
   email: string;
   fieldErrors: FieldError[];
-  idapiBaseUrl: string;
 };
 
-/* eslint-disable functional/immutable-data */
-// explicit use of mutable ref
-// useRefState is used to provide access to up-to-date component state inside promises
-function useRefState<T>(
-  initial: T,
-): [T, Dispatch<SetStateAction<T>>, MutableRefObject<T>] {
-  const [state, setState] = useState(initial);
-  const stateRef = useRef(state);
-  useEffect(() => {
-    stateRef.current = state;
-  }, [state]);
-  return [state, setState, stateRef];
+enum Validation {
+  SUCCESS,
+  FAILURE,
+  ERROR,
 }
-/* eslint-enable functional/immutable-data */
-
-const checkIfBreached = AwesomeDebouncePromise(
-  (password: string): Promise<PasswordValidationResult> => {
-    const hashedPassword = sha1(password);
-    const firstFive = hashedPassword.substr(0, 5);
-    const remainingHash = hashedPassword.substr(5, hashedPassword.length);
-    return fetch(`https://api.pwnedpasswords.com/range/${firstFive}`)
-      .then((r) =>
-        r.text().then((results) => {
-          if (results.includes(remainingHash.toUpperCase())) {
-            return PasswordValidationResult.COMMON_PASSWORD;
-          } else {
-            return PasswordValidationResult.VALID_PASSWORD;
-          }
-        }),
-      )
-      .catch(() => {
-        return PasswordValidationResult.VALID_PASSWORD;
-      });
-  },
-  300,
-);
 
-const validatePassword = async (
-  idapiBaseUrl: string,
-  password: string,
-): Promise<PasswordValidationResult> => {
-  const lengthResult = validatePasswordLength(password);
-  if (lengthResult !== PasswordValidationResult.VALID_PASSWORD) {
-    return Promise.resolve(lengthResult);
-  }
-
-  return checkIfBreached(password);
-};
+const ValidationSymbol = ({ result }: { result: Validation }) => {
```

Comment:
it might be easier to split this into 3 separate components which share baseStyles

Reply:
Yeah, I think you're right. After the changes I've been making this would be the next logic thing to do here.

Reply:
Done